rvps/src/reference_value.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! reference value for RVPS
+
+use chrono::{DateTime, NaiveDateTime, Utc};
+use serde::{Deserialize, Deserializer, Serialize};
+
+/// Default version of ReferenceValue
+pub const REFERENCE_VALUE_VERSION: &str = "0.1";
+
+/// A HashValuePair stores a hash algorithm name
+/// and relative artifact's hash value due to
+/// the algorithm.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct HashValuePair {
+    alg: String,
+    value: String,
+}
+
+impl HashValuePair {
+    pub fn new(alg: String, value: String) -> Self {
+        Self { alg, value }
+    }
+
+    pub fn alg(&self) -> &String {
+        &self.alg
+    }
+
+    pub fn value(&self) -> &String {
+        &self.value
+    }
+}
+
+/// Helper to deserialize an expired time
+fn primitive_date_time_from_str<'de, D: Deserializer<'de>>(
+    d: D,
+) -> Result<DateTime<Utc>, D::Error> {
+    let s: Option<String> = Deserialize::deserialize(d)?;
+    if s.is_none() {
+        return Err(serde::de::Error::invalid_length(0, &"<TIME>"));
+    }
+    let s = s.unwrap();
```

Comment:
This would be cleaner using `map_err(...)?` or similar.

Reply:
Sure!